apps/entity-service/src/entities/processors/media.processor.ts

```diff
@@ -0,0 +1,230 @@
+import {
+  Media,
+  Project,
+  ProjectReport,
+  Site,
+  SiteReport,
+  Nursery,
+  User,
+  NurseryReport
+} from "@terramatch-microservices/database/entities";
+import { MediaDto } from "../dto/media.dto";
+import { EntityModel, EntityType, EntityClass } from "@terramatch-microservices/database/constants/entities";
+import { AssociationProcessor } from "./association-processor";
+import { MediaQueryDto } from "../dto/media-query.dto";
+import { EntitiesService } from "../entities.service";
+import { DocumentBuilder, getDtoType, getStableRequestQuery } from "@terramatch-microservices/common/util";
+import { col, fn, Op, Sequelize } from "sequelize";
+import { UserDto } from "@terramatch-microservices/common/dto/user.dto";
+
+export class MediaProcessor extends AssociationProcessor<Media, MediaDto> {
+  readonly DTO = MediaDto;
+
+  constructor(
+    protected readonly entityType: EntityType,
+    protected readonly entityUuid: string,
+    protected readonly entityModelClass: EntityClass<EntityModel>,
+    protected readonly entitiesService: EntitiesService
+  ) {
+    super(entityType, entityUuid, entityModelClass, entitiesService);
+  }
+
+  private async getSiteModels(site: Site) {
+    const models = [{ modelType: this.entityModelClass.LARAVEL_TYPE, ids: [site.id] }];
+
+    const siteReports = await SiteReport.findAll({ where: { siteId: site.id }, attributes: ["id"] });
```

Comment:
Why is fetching a site's media also fetching the media for the site report? That sort of pattern seems to be greatly complicating this processor. It's also not super intuitive to me. It seems like if I want the media for a site report, I'll send another request to get that site report's media. If it's super common to need all the media for a given site's reports on the site's page, then we should explore supporting that with a sideloading mechanism, like I did for the my projects page: https://github.com/wri/terramatch-microservices/blob/22c0727cdd850d169d02e9bbe14d1a26dba00017/apps/entity-service/src/entities/processors/project.processor.ts#L101-L112. If it seems from the FE usage like this is something we really need, I can help you add sideloading to the association processor pattern.

Reply:
This one is tricky, because I need to be able to search and sort based on date of all records (for example by createdAt).
Watching the current side load I don't think this one fits.
Also this is the current behavior on v2

Reply:
There are lots of bad behaviors on v2 that we don't want to replicate in v3. Which FE page is this needed for? Maybe I can take a look and suggest some alternative to product. It's totally possible the current behavior is not necessary and while we're moving things to v3 it's always important to identify and reduce bad code / API patterns.

Reply:
`project/9db2495f-bee1-4323-97ac-c86a3a4e0283?tab=gallery`
`site/553f4d72-73a0-47b5-bc30-1389d243c5ce?tab=gallery`
Basically all the gallery tabs for PD views